code/modules/crew_objectives/civilian_objectives.dm

```diff
@@ -0,0 +1,223 @@
+			if(H.z == ZLEVEL_STATION_PRIMARY || SSshuttle.emergency.shuttle_areas[get_area(H)])
+				return FALSE
+	return TRUE
+
+/datum/objective/crew/clean //ported from old Hippie
+	var/list/areas = list()
+	var/hardmode = FALSE
+	explanation_text = "Ensure sure that (Yo, something broke. Yell about this in citadels devlopmeent discussion channel.) remain spotless at the end of the shift."
+	jobs = "janitor"
+
+/datum/objective/crew/clean/New()
+	. = ..()
+	if(prob(1))
+		hardmode = TRUE
+	var/list/blacklistnormal = list(typesof(/area/space) - typesof(/area/lavaland) - typesof(/area/mine) - typesof(/area/ai_monitored/turret_protected) - typesof(/area/tcommsat))
+	var/list/blacklisthard = list(typesof(/area/lavaland) - typesof(/area/mine))
+	var/list/possibleareas = list()
+	if(hardmode)
+		possibleareas = GLOB.teleportlocs - /area - blacklisthard
+	else
+		possibleareas = GLOB.teleportlocs - /area - blacklistnormal
+	for(var/i in 1 to rand(1,6))
+		areas |= pick_n_take(possibleareas)
+	update_explanation_text()
+
+/datum/objective/crew/clean/update_explanation_text()
+	. = ..()
+	explanation_text = "Ensure that the"
+	for(var/i in 1 to areas.len)
+		var/area/A = areas[i]
+		explanation_text += " [A]"
+		if(i != areas.len && areas.len >= 3)
+			explanation_text += ","
+		if(i == areas.len - 1)
+			explanation_text += "and"
+	explanation_text += " [(areas.len ==1) ? "is completely" : "are [(areas.len == 2) ? "completely" : "all"]"] clean at the end of the shift."
+	if(hardmode)
+		explanation_text += " Chop-chop."
+
+/datum/objective/crew/clean/check_completion()
+	for(var/area/A in areas)
+		for(var/obj/effect/decal/cleanable/C in area_contents(A))
+			return FALSE
+	return TRUE
+
+/datum/objective/crew/slipster //ported from old Hippie with adjustments
+	explanation_text = "Slip at least (Yell on GitHub if you see this) different people with your PDA, and have it on you at the end of the shift."
+	jobs = "clown"
+
+/datum/objective/crew/slipster/New()
+	. = ..()
+	target_amount = rand(5, 20)
+	update_explanation_text()
+
+/datum/objective/crew/slipster/update_explanation_text()
+	. = ..()
+	explanation_text = "Slip at least [target_amount] different people with your PDA, and have it on you at the end of the shift."
+
+/datum/objective/crew/slipster/check_completion()
+	var/list/uniqueslips = list()
+	if(owner && owner.current)
+		for(var/obj/item/device/pda/clown/PDA in owner.current.get_contents())
+			for(var/mob/living/carbon/human/H in PDA.slipvictims)
+				uniqueslips |= H
+	if(uniqueslips.len >= target_amount)
+		return TRUE
+	else
+		return FALSE
+
+/datum/objective/crew/vow //ported from old Hippie
+	explanation_text = "Never break your vow of silence."
+	jobs = "mime"
+
+/datum/objective/crew/vow/check_completion()
+	if(owner && owner.current)
+		var/list/say_log = owner.current.logging[INDIVIDUAL_SAY_LOG]
+		if(say_log.len > 0)
+			return FALSE
+	return TRUE
+
+/datum/objective/crew/nullrod
+	explanation_text = "Don't lose your holy rod."
```

Comment:
You can guess what this objective also checked for on citadel if the chaplain was male.

Reply:
probably needs changed to null rod or holy artefact